jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/property/DefaultWebProperties.java

```diff
@@ -83,4 +89,41 @@ public Boolean value(WebPropertyContext ctx) {
         };
     }
 
+    @Bean
+    public WebProperty<Boolean> displayedOnScreenWebProperty() {
+        return new AbstractWebProperty<Boolean>() {
+            @Override
+            public String name() {
+                return "DISPLAYED_ON_SCREEN";
+            }
+
+            @Override
+            public Boolean value(WebPropertyContext ctx) {
+                try {
+                    WebElement element = findElement(ctx);
+                    boolean isVisible = element.isDisplayed();
+                    if (driver instanceof JavascriptExecutor) {
+                        boolean visibleOnScreen = (Boolean) ((JavascriptExecutor) driver).executeScript(
+                            "var elem = arguments[0],                 " +
+                                "  box = elem.getBoundingClientRect(),    " +
+                                "  cx = box.left + box.width / 2,         " +
+                                "  cy = box.top + box.height / 2,         " +
+                                "  e = document.elementFromPoint(cx, cy); " +
+                                "for (; e; e = e.parentElement) {         " +
+                                "  if (e === elem)                        " +
+                                "    return true;                         " +
+                                "}                                        " +
+                                "return false;                            "
+                            , element);
+                        return (isVisible && visibleOnScreen);
```

Comment:
minor: I'd use some form of "display" in the naming of local variables when the whole property/class use the name DISPLAY